docs/SUMMARY.md

```diff
@@ -10,38 +15,8 @@
     - [Profiling with VTune](./examples-profiling-vtune.md)
     - [Profiling with samply](./examples-profiling-samply.md)
     - [Cross-platform Profiling](./examples-profiling-guest.md)
-  - [Embedding in Rust](./examples-rust-embed.md)
-    - [Hello, world!](./examples-rust-hello-world.md)
-    - [Calculating the GCD](./examples-rust-gcd.md)
-    - [Using Linear Memory](./examples-rust-memory.md)
-    - [WASI](./examples-rust-wasi.md)
-    - [Linking Modules](./examples-rust-linking.md)
-    - [Debugging](./examples-rust-debugging.md)
-    - [Core Dumps](./examples-rust-core-dumps.md)
-    - [Using Multi-Value](./examples-rust-multi-value.md)
-  - [Embedding in C](./examples-c-embed.md)
-    - [Hello, World!](./examples-c-hello-world.md)
-    - [Calculating the GCD](./examples-c-gcd.md)
-    - [Using Linear Memory](./examples-c-memory.md)
-    - [WASI](./examples-c-wasi.md)
-    - [Linking Modules](./examples-c-linking.md)
-    - [Debugging](./examples-c-debugging.md)
-    - [Using Multi-Value](./examples-c-multi-value.md)
```

Comment:
We still check that the associated examples compile in CI, so I'm not sure that getting rid of these is actually warranted? I feel like examples are the most useful form of non-API documentation and I think it would be a shame to just remove all of these. Maybe just keep the Rust versions?

The GCD examples, framed as GCD examples, don't really seem that useful but it is a simple example of the typed function API, which I think is valuable. Maybe we could reframe / retitle it or something?

The memory example seems useful.

The WASI example is dated. Definitely better places to go as far as toolchain docs go. But I think there is room for an example of how you call `add_to_linker` and all that with Wasmtime to *consume* WASI components (and probably also worth mentioning legazy preview1 stuff for a while, fwiw). We get a lot of issues filed by people not realizing they need to call `add_to_linker` and I think having an example to point people at would be good.

The linking example is dated. Subsumed by component model stuff. But we might want to have an example of using the `Linker` and defining custom functions to be available for import.

Language-specific debugging example seems redundant with the debugging section of the docs, although that section links to the Rust debugging bit. We should inline that into the debugging section (if you haven't already, haven't looked in detail at the rest of the changes yet).

Same thing for the language-specific core dumps bit.

The multi-value example seems like it is still generally useful.